keps/2936-local-queue-defaulting/README.md

```diff
@@ -0,0 +1,63 @@
+# KEP-2936: LocalQueue defaulting
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a defaulting LocalQueue mechanism for the jobs that don't have
+LocalQueue name annotation specified.
+
+## Motivation
+
+Simplify user experience with Kueue.
+
+### Goals
+
+- default LocalQueue for jobs that don't specify LocalQueue name annotation.
+
+### Non-Goals
+
+- create a default LocalQueue.
+
+## Proposal
+
+If the Job doesn't have annotation for LocalQueue in its Spec and the LocalQueue
+with name `default` is present in Job's namespace and KueueDefaulting feature gate
+is enabled, the `kueue.x-k8s.io/queue-name:default` anotation will be added to the Job.
+
+### User Stories
+
+#### Story 1
+
+Administrator createds a single `default` LocalQueue per namespace. Users doesn't need to specify
+the LocalQueue annotation in order for workload to be scheduled.
+
+### Notes/Constraints/Caveats (Optional)
+
+manageJobsWithoutQueueName doesn't work with this feature, since Kueue add the LocalQueue
+annotation.
+
+### Risks and Mitigations
+
+## Design Details
```

Comment:
Instead of using the reserved `default` name for the default localQueue, could we scrape the default queue-name from the cluster?

For example, we might want to prepare the dedicated `.spec.default: true|false` field in the LocalQueue, and then the webhook looks into the default queue-name when the webhook insert the queue-name to Jobs.

This idea is similar to storageClass.

Reply:
>.spec.default: true|false field in the LocalQueue

Having default field in the API brings it owns challenge, I mentioned it earlier in the issue: https://github.com/kubernetes-sigs/kueue/issues/2936#issuecomment-2478698423, i.e. validating that only one LQ is default.

Since the goal is to simplify workflow, it seems having `default` LQ is simple and explicitly. 
However I'm not sure about having a defaulting under the feature gate. In case the feature goes to GA and users are not familiar with the defaulting mechanism, they may create the LQ with default name and jobs without queue label and expecting a different behaviour. 
Maybe instead of feature gate it should be a part of kueue config.

Reply:
>This idea is similar to storageClass.

StorageClass allows multiple defaults and the latest default is an actual default. I also considered this as an alternative (will add it to alternatives section).

I'm thinking about use case. 1) In the case described in the issue, when admin creates only one LQ that is supposed to use both solutions (`default` LQ and spec.default=true) would work fine. 
2) In case when there many LQs and one of them is default it seems having `default` LQ is better, because it easier to find a `default` LQ among hundreds of LQs in namespace and modify (not sure if it's possible, maybe recreate) in case of misconfiguration.

Reply:
That makes sense.
I think that the custom default LQ name is still valid request, but it seems that the customization will require an additional mechanism like comparing creationTimestamp between multiple defaults LQs.

So, I agree with the fixed default LQ name, "default" so that we can start this feature as a minimum.
After we obtain some feedbacks for the custom default LQ name, we can revisit here, and consider the customization.